sihl/core/ctx.ml

```diff
@@ -12,3 +12,15 @@ let find key ctx = Hmap.find key ctx.map
 let remove key ctx = { ctx with map = Hmap.rem key ctx.map }
 let create_key = Hmap.Key.create
 let id ctx = ctx.id
+let sexp_of_t t = Sexplib.Std.sexp_of_string t.id
+let counter = ref 0
+
+let create ?id () =
```

Comment:
Why can `id` be set here?

Reply:
Because the HTTP layer have to be able to set it